nifi-nar-bundles/nifi-provenance-repository-bundle/nifi-persistent-provenance-repository/src/main/java/org/apache/nifi/provenance/PersistentProvenanceRepository.java

```diff
@@ -521,6 +522,23 @@ public RepositoryConfiguration getConfiguration() {
         return configuration;
     }
 
+    @Override
+    public Set<String> getContainerNames() {
+        return new HashSet<>(configuration.getStorageDirectories().keySet());
+    }
+
+    @Override
+    public long getContainerCapacity(final String containerName) throws IOException {
+        final Path path = configuration.getStorageDirectories().get(containerName).toPath();
```

Comment:
`FileUtils.getContainerCapacity` verifies the containerName is non null. However, if the specified container is unknown to this repository it will fail here prior to invoking the `FileUtils`. This issue exists in a number of places in this PR.